## Libraries

In [3]:
import pandas as pd


Only pandas library was used in this section.

## Data Wrangling

In [17]:
cols = ['Nome do Arquivo', 'iCodDocumento']
xls = pd.read_csv('../03_spreadsheets/xls-dataset.csv', sep = ';', usecols=cols)
xls.head()

Nome do Arquivo  iCodDocumento
0         03000000012017_6622932_inicial_lorena.pdf            5070
1                   16000000032017_6626319_p.i..pdf            5070
2                   16000000042017_6626401_p.i..pdf            5070
3   11000000012017_6626489_00122055920165030077.pdf            5070
4  04000000012017_6627051_NP00017480420165100002G...           5083

In [5]:
cols=['Nome do Arquivo','Palavras-chave']
api = pd.read_csv('../02_azure_search/api-dataset.csv', sep = ';', usecols=cols)
api.head()

Palavras-chave  \
0  e o Palace e pior circunstancias e e sobre Mut...   
1  opal op op ap orminr ap of ap nuvo op e sueuly...   
2  para que que alterou o horas que os consideran...   
3  DA CLT e o funcionamento DA SÚMULA FEDERAL DA ...   
4  lotacao e eleicoes e bancos e Mercado e CAIXA ...   

                                     Nome do Arquivo  
0  02000008202017_7611299_20170413_FEDERAL_SEGURO...  
1  02000008202017_7611303_20170413_FEDERAL_SEGURO...  
2  02000000722017_6983943_PETICAO_INICIAL_ANA_LUC...  
3                      02000001342017_6874716_PI.pdf  
4           02001001382017_6873843_Documento(15).pdf

The [xls-dataset](../03_spreadsheets/xls-dataset.csv "xls dataset") was loaded into a pandas DataFrame and a preview was presented. Same process was applied to the [api-dataset](../02_azure_search/api-dataset.csv "api dataset").

In [6]:
xls.set_index('Nome do Arquivo', inplace=True)
api.set_index('Nome do Arquivo', inplace=True)

In [7]:
print(xls.shape)
print(api.shape)

(2126, 1)
(600, 1)


'Nome do Arquivo' was setted as index to both datasets and their dimensions were shown. Xls dataset had 2126 lines and 'iCodDocumento' column, whereas Api dataset had 600 lines and 'Palavras-chave' column.

In [8]:
api.index.difference(xls.index)

Index(['02000000122017_7284093_PETICAO_INICIAL_00023054620165110007.pdf',
       '02000000182017_6739022_RT_00009552320165110201_DAYANA_DE_SOUZA_CARVALHO.pdf',
       '02000000212017_6758941_PETICAO_INICIAL_RENATA.pdf',
       '02000000222017_6758570_PETICAO_INICIAL__ROGERIO_PESSOA_DE_MELO.pdf',
       '02000000302017_6761978_Mandado_de_Intimacao_-_Proc_16374-70.2016.4.01.3200.pdf',
       '02000000312017_6778931_PETICAO_INICIAL_MARCOS_ROBERTO_DA_CUNHA.pdf',
       '02000000312017_7201808_1.INICIAL.zip',
       '02000000352017_6774071_20170119_MANDADO_DE_CITACAO_-_181890520164013200.pdf',
       '02000000452017_6798683_RT_00023236120165110009_ADRIANA_SAID_NEGREIROS_MASCARENHAS_PONTES.pdf',
       '02000000532017_6809593_20170125_-_PROCESSO_INTEGRAL_-_OSVANDO_REIS_QUEIROZ_0002532-03.2016.5.11.0018.pdf',
       ...
       '02001002112017_7020899_Documento(29).pdf',
       '02001002112017_7020900_Documento(30).pdf',
       '02001002112017_7020901_Documento(31).pdf',
       '02001002112017

In [9]:
xls.index = xls.index.str.strip()
api.index.difference(xls.index)

Index(['02000000532017_6809593_20170125_-_PROCESSO_INTEGRAL_-_OSVANDO_REIS_QUEIROZ_0002532-03.2016.5.11.0018.pdf',
       '02000005792017_7294356_PETICAO_INICIAL_JURANDIR_BARBOSA_DA_MOTA_-_0000332-83.2017.5.11.0019.pdf',
       '02000005802017_7295032_20170316_PETICAO_INICIAL_LEOMAR_SOUZA_DE_LISBOA_-_0000581-86.2016.5.11.0401.pdf',
       '02000008202017_7611300_20170413_FEDERAL_SEGUROS_-_00820_2017_-_0224726-83.2011.8.04.0001_18.pdf',
       '02000008202017_7611301_20170413_FEDERAL_SEGUROS_-_00820_2017_-_0224726-83.2011.8.04.0001_17.pdf',
       '02000008202017_7611302_20170413_FEDERAL_SEGUROS_-_00820_2017_-_0224726-83.2011.8.04.0001_16.pdf',
       '02000008202017_7611303_20170413_FEDERAL_SEGUROS_-_00820_2017_-_0224726-83.2011.8.04.0001_15.pdf',
       '02000008202017_7611304_20170413_FEDERAL_SEGUROS_-_00820_2017_-_0224726-83.2011.8.04.0001_14.pdf',
       '02000008202017_7611305_20170413_FEDERAL_SEGUROS_-_00820_2017_-_0224726-83.2011.8.04.0001_13.pdf',
       '02000008202017_7611306

In [10]:
xls.index = xls.index.str.replace(' /V /', '', regex=False)
api.index.difference(xls.index)

Index(['02000000532017_6809593_20170125_-_PROCESSO_INTEGRAL_-_OSVANDO_REIS_QUEIROZ_0002532-03.2016.5.11.0018.pdf', '02000005802017_7295032_20170316_PETICAO_INICIAL_LEOMAR_SOUZA_DE_LISBOA_-_0000581-86.2016.5.11.0401.pdf'], dtype='object', name='Nome do Arquivo')

In [11]:
row = [
    '02000000532017_6809593_20170125_-_PROCESSO_INTEGRAL_-_OSVANDO_REIS_QUEIROZ_0002532-03.2016.5.11.0018',
    '02000005802017_7295032_20170316_PETICAO_INICIAL_LEOMAR_SOUZA_DE_LISBOA_-_0000581-86.2016.5.11.0401.p'
]
fix = [
    '02000000532017_6809593_20170125_-_PROCESSO_INTEGRAL_-_OSVANDO_REIS_QUEIROZ_0002532-03.2016.5.11.0018.pdf',
    '02000005802017_7295032_20170316_PETICAO_INICIAL_LEOMAR_SOUZA_DE_LISBOA_-_0000581-86.2016.5.11.0401.pdf'
]
xls.rename(index={row[i]: fix[i] for i in range(len(row))}, inplace=True)
api.index.difference(xls.index)

Index([], dtype='object', name='Nome do Arquivo')

The differences between indexes were studied. Some problems were found in the [provided worksheet](../01_source/Rela‡Æo%20de%20expedientes%20para%20capturar%20arquivos%20no%20GED_resposta.xls "").  
The first analysis revealed that some files names had blanks spaces in the end, so stip function was applied to 'Nome do Arquivo' on xls dataset.  
A second analysis identified ' /V /' in the end of some names, a string replacement was used to remove this string from the names.  
The third analysis were more specific, where two names were incomplete, for them the data were manually defined and fixed.

## Merge Datasets

In [12]:
data = pd.merge(xls, api, left_index=True, right_index=True)

In [13]:
data.shape

(600, 2)

In [14]:
data.head()

iCodDocumento  \
Nome do Arquivo                                                     
03000000012017_6622932_inicial_lorena.pdf                    5070   
03000001632017_6667276_inicial_valquiria.pdf                 5070   
02001000312017_6699679_20160112_MANDADO_DE_CITA...           5070   
02000000182017_6739022_RT_00009552320165110201_...           5070   
02000000222017_6758570_PETICAO_INICIAL__ROGERIO...           5070   

                                                                                       Palavras-chave  
Nome do Arquivo                                                                                        
03000000012017_6622932_inicial_lorena.pdf           DA NECESSIDADE E TITULAR  DA DA PRETERIÇÃO EM ...  
03000001632017_6667276_inicial_valquiria.pdf        acionada que O direito que assiste Indenização...  
02001000312017_6699679_20160112_MANDADO_DE_CITA...  e que pendem Regiao - e Trabalhasta e quitados...  
02000000182017_6739022_RT_00009552320165110201_...  próprio e o DA MULTA o que o fez DA COMPETÊNCI...  
02000000222017_6758570_PETICAO_INICIAL__ROGERIO...  que quebra caixa não caixa que eram caixa para...

In [15]:
data.to_csv('dataset.csv', sep = ';')

After Data Wrangling process the datasets were merged using 'Nome do Arquivo' as key. As a result, a new dataset with 600 lines and 2 columns (one of each input) were created, previewed and saved into a [consolidated dataset on a csv file](./dataset.csv). It is noteworthy that all comum files names were merged sucessfully.